packages/opentelemetry-core/src/platform/node/environment.ts

```diff
@@ -29,6 +30,92 @@ export function getEnv(): Required<ENVIRONMENT> {
   return Object.assign({}, DEFAULT_ENVIRONMENT, processEnv);
 }
 
+/**
+ * Retrieves a number from an environment variable.
+ * - Trims leading and trailing whitespace.
+ * - Returns `undefined` if the environment variable is empty, unset, contains only whitespace, or is not a number.
+ * - Returns a number in all other cases.
+ *
+ * @param {string} key - The name of the environment variable to retrieve.
+ * @returns {number | undefined} - The number value or `undefined`.
+ */
+export function getNumberFromEnv(key: string): number | undefined {
+  const raw = process.env[key]?.trim();
+  if (raw == null || raw === '') {
+    return undefined;
+  }
+
+  const value = Number(raw);
+  if (isNaN(value)) {
+    diag.warn(
+      `Unknown value ${raw} for ${key}, expected a number, using defaults`
```

Comment:
This `.trim()` is new. Do you want to add it here?
If it is only for the `=== ''` empty check, the `?.trim()` call could get moved there. `Number(s)` will ignore whitespace. One slight benefit is that the diag.warn could show that actual whitespace in the raw value. E.g. something like this

```suggestion
  const raw = process.env[key];
  if (raw == null || raw?.trim() === '') {
    return undefined;
  }

  const value = Number(raw);
  if (isNaN(value)) {
    diag.warn(
      `Unknown value ${util.inspect(raw)} for ${key}, expected a number, using defaults`
```

I'm being a bit paranoid using `util.inspect()`. It'll do a decent job for *whatever* type of garbage happens to be set on process.env.  The quoting would help debug a weird case where a user messed up quoting and set:

```
export OTEL_BSP_EXPORT_TIMEOUT='"42"'
```

Anyway, this is a nit. :)

Reply:
oh, nice, added in https://github.com/open-telemetry/opentelemetry-js/pull/5443/commits/ffe7e1e4389b5086bd4a4952d51b0ba4b6c0038e 🙂